vllm/model_executor/models/transformers.py

```diff
@@ -166,6 +169,28 @@ def replace_linear_class(
     )
 
 
+def replace_rms_norm_class(rms_norm: nn.Module) -> RMSNorm:
+    """Replace a Transformers RMSNorm with vLLM's RMSNorm.
+
+    This method assumes:
+    - Weight is stored as `weight`.
+    - Epsilon is stored as `eps` or `variance_epsilon`.
+    - `with_scale` indicates whether the layer has a weight (Gemma3n only).
+    - `var_hidden_size` is only ever used for Intern vision encoder.
+    """
+    weight = getattr(rms_norm, "weight", None)
+    weight: Optional[torch.Tensor] = getattr(weight, "data", weight)
+    # Return early if weight not found
+    if weight is None:
+        return rms_norm
```

Comment:
Why not use vllm's `RMSNorm` without weight (AFAIK that's supported)

Reply:
Good point, if there's no weight I can likely just use the vLLM op anyway